sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala

```diff
@@ -86,10 +89,28 @@ case class HadoopFsRelation(
   }
 
   override def sizeInBytes: Long = {
-    val compressionFactor = sqlContext.conf.fileCompressionFactor
-    (location.sizeInBytes * compressionFactor).toLong
+    val isCatalogFileIndex = location.isInstanceOf[CatalogFileIndex]
+    if (sparkSession.sessionState.conf.fallBackToHdfsForStatsEnabled && isCatalogFileIndex) {
+      sizeInBytesFallBackToHdfs
+    } else {
+      val compressionFactor = sqlContext.conf.fileCompressionFactor
+      (location.sizeInBytes * compressionFactor).toLong
+    }
   }
 
+  private def sizeInBytesFallBackToHdfs: Long = {
```

Comment:
Rather than repeat the 'compression' part here, you could inline this method, `return` from the try block, and ignore the exception, falling through to a default return value with the 'compression' logic

Reply:
I have updated the PR, based on the above comments